### ISSUE TITLE
visualmetrics with "carousel" element 

### ISSUE THREAD

**Issue by matti** [sitespeedio/sitespeed.io#2759]
Visualmetrics last visual change for https://www.kaalimato.com/ is ~25s because the carousel keeps changing the page - is there any way to mitigate this?

**Comment by soulgalore** [comment]
You can choose when to end your test with https://www.sitespeed.io/documentation/sitespeed.io/browsers/#choose-when-to-end-your-test

**Comment by matti** [comment]
hmm, but wouldn't a better solution be some sort of `--visuallyCompleteIfNotChangesDuring 2.0` ?

**Comment by matti** [comment]
Like that would handle it for *every* page/domain?

**Comment by soulgalore** [comment]
The video doesn't decide when to end, it just record the screen. My guess in your case that the loadEventEnd hasn't fired (not all request/responses are done). What happens on screen isn't the same as which requests being handled by the browser.

**Comment by matti** [comment]
Yeah, the page keeps loading with every single carousel page.  I still think it should be possible to not write custom handler every page. I'll think about this.

**Comment by soulgalore** [comment]
It works like this: You can run your own JS that decides when to end. But by default WebDriver gives back control after loadEventEnd happens, so you need to change the `--pageLoadStrategy `so you get into control direct after the navigation.

**Comment by matti** [comment]
![Page_timing_metrics_-_Grafana](https://user-images.githubusercontent.com/3893/68413770-92a1fb00-0197-11ea-81bc-2fc7f0eb2601.png)

But loadEventEnd happens at 1.54s, "fullyLoaded" happens at 30s (timeout?) - So I don't think it's about pageLoadStrategy -- the control is returned with normal strategy?

**Comment by soulgalore** [comment]
Then I would guess that we stop the video after we collected all the JS metrics, instead of do it before. Need to check the implementation.

**Comment by soulgalore** [comment]
I think we could change it but it needs some changing/splitting the code. Today we need to know the URL from the page when we stop the video, that need to be fixed. Then we could close the video right before we stop collect metrics using JS. But it will need some testing to make sure we don't introduce new things. Moving this to Browsertime since its there it needs to be implemented.
